backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/graphicpromotion/internal/BpkGraphicPromoImpl.kt

```diff
@@ -0,0 +1,271 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.graphicpromotion.Sponsor
+import net.skyscanner.backpack.compose.graphicpromotion.Variant
+import net.skyscanner.backpack.compose.graphicpromotion.VerticalAlignment
+import net.skyscanner.backpack.compose.overlay.BpkOverlay
+import net.skyscanner.backpack.compose.overlay.BpkOverlayType
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@Composable
+internal fun BpkGraphicPromoImpl(
+    headline: String,
+    image: @Composable BoxScope.() -> Unit,
+    modifier: Modifier = Modifier,
+    kicker: String? = null,
+    subHeadline: String? = null,
+    overlayType: BpkOverlayType? = null,
+    variant: Variant = Variant.OnDark,
+    verticalAlignment: VerticalAlignment = VerticalAlignment.Top,
+    sponsor: Sponsor? = null,
+    sponsorLogo: @Composable () -> Unit? = {},
+    tapAction: () -> Unit = {},
+) {
+    val roundedCornerShape = RoundedCornerShape(BpkBorderRadius.Md)
+    val contentDescription = listOfNotNull(kicker, headline, subHeadline, sponsor?.accessibilityLabel)
+        .joinToString(separator = ", ")
+
+    val interactionSource = remember { MutableInteractionSource() }
+    val isPressed by interactionSource.collectIsPressedAsState()
+
+    Box(
+        modifier = modifier
+            .aspectRatio(RATIO_PORTRAIT)
+            .clip(roundedCornerShape)
+            .clickable(
+                interactionSource = interactionSource,
+                indication = null,
+                onClick = tapAction,
+            )
+            .semantics(mergeDescendants = true) {
+                role = Role.Button
+                this.contentDescription = contentDescription
+            },
+    ) {
+        overlayType?.let {
+            BpkOverlay(
+                modifier = Modifier.fillMaxSize(),
```

Comment:
If you are using a component within the BoxScope, there's machParentSize() modifier available. It behaves the same as fillMaxSize(), except it doesn't force the parent to stretch. Let's use it instead here and in the other places.